### PR TITLE
Refactor HostInfo

### PR DESCRIPTION
**Pull request by Ingvord**
Using JTango-9.3.3-SNAPSHOT I can not subscribe to an event.

Server output:

```
Host name/address cannot be determined !
```

Looks like this is due to this method (HostInfo#checkInetAddress)

Basically it fails to set proper host info when host name equals to IP address... In overall HostInfo implementation seems strange to me:

* singleton
* actual info depends on NI order (the last one always win, which might not be the desired behavior)
* uses undocumented trace flag instead of logging framework

This PR is to refactor HostInfo. Possibly remove it at all.

### CODE REVIEW

File: common/src/main/java/fr/esrf/TangoApi/HostInfo.java [hunk 1]
```diff
@@ -30,159 +30,107 @@
...
+    private HostInfo() {
+    }
+
+    /**
+     * @return all Ip (4&6) available on this host
+     * @throws DevFailed
+     */
+    public static Iterable<String> getIpAddresses() throws DevFailed {
+        List<String> result;
+        if (OAI_ADDR != null && !OAI_ADDR.isEmpty()) {
```
Comment (Ingvord) [inline common/src/main/java/fr/esrf/TangoApi/HostInfo.java:76]: I am not sure about this one. I would return all possible addresses in this method (remove this check at all). This check can be moved to methods related to single address, i.e. _getAddress_ etc

**Comment by Ingvord** [comment]
Ready for review!

@Pascal-Verdier , @gwen-soleil , @bourtemb your input is very welcome!

Note due to usages of _HostInfo_ this PR contains also refacctoring of _EventManager_ - it now binds to every IPv4 address available on the host + _AccessProxy_ has been modified (logic should be the same)

This PR is scheduled for merge in a week (10/08/17) if no severe input will be provided

**Comment by Pascal-Verdier** [comment]
Hi 
I did some tests with your changes on HostInfo.java
It seems to have same behavior in our control system.
I suppose you can merge.

Cheers
Pascal

**Comment by Ingvord** [comment]
@Pascal-Verdier thanks!

**Pull request merged**